x/gamm/pool-models/stableswap/pool.go

```diff
@@ -416,6 +416,36 @@ func (p *Pool) SetScalingFactors(ctx sdk.Context, scalingFactors []uint64, sende
 	return nil
 }
 
+// GetMaximalNoSwapLPAmount returns the coins(lp liquidity) needed to get the specified amount of shares in the pool.
+// Steps to getting the needed lp liquidity coins needed for the share of the pools are
+// 1. calculate how much percent of the pool does given share account for(# of input shares / # of current total shares)
+// 2. since we know how much % of the pool we want, iterate through all pool liquidity to calculate how much coins we need for
+// each pool asset.
+func (p Pool) GetMaximalNoSwapLPAmount(ctx sdk.Context, shareOutAmount sdk.Int) (neededLpLiquidity sdk.Coins, err error) {
```

Comment:
Why are we duplicating the implementations? If both balancer and stableswap need to have essentially the same logic, keeping it as function makes sense to me